frontend/src/modules/Worksheets/views/WorksheetView.js

```diff
@@ -181,11 +181,18 @@ const WorksheetView = () => {
           response.data.searchEnvironmentDataItems.nodes.map((d) => ({
             datasetUri: d.datasetUri,
             value: d.datasetUri,
-            label: `${d.GlueDatabaseName}_shared_${d.shareUri}`,
-            GlueDatabaseName:
-              `${d.GlueDatabaseName}_shared_${d.shareUri}`.substring(0, 254),
+            label: d.sharedGlueDatabaseName,
+            GlueDatabaseName: d.sharedGlueDatabaseName,
             environmentUri: d.environmentUri
           }));
+        // Remove duplicates based on GlueDatabaseName
+        sharedWithDatabases = sharedWithDatabases.filter(
```

Comment:
Could we potentially remove this duplication in the query itself ? `dataall.modules.dataset_sharing.db.share_object_repositories.ShareObjectRepository.paginate_shared_datasets` in here

Reply:
Not sure if we want to remove duplication of DB names on the backend because this query `searchEnvironmentDataItems` is also used at `frontend/src/modules/Environments/components/EnvironmentSharedDatasets.js` where we show each share item that is shared with the given environment and for what respective team in that environment (in this case we would want to show all shares duplicate dbs or not)

Another note (out of scope for this PR): We currently do not handle shared items of type `S3Bucket` for the table mentioned above ...

Reply:
Oh okay I get it. Thanks @noah-paige . 

I didn't quite get your what you meant by the note and which table you are mentioning

Reply:
There are 2 reasons not to remove the duplicates, one is that the query is used in other places as Noah pointed out and the second reason was that it is difficult to predict which shares are using the old method and the new method, and that is retrieved in the resolver that is a field of the type. Once we deprecate the old type of share names I think we can redesign the remove-duplicates and the separation between plot the shared items in environment and list the shared databases in Worksheets. I tried implementing a new resolver (check the commits), but I found it a bit complex to fulfill the purpose of this PR

Reply:
Yeah makes sense. Thanks @dlpzx